x/lockup/keeper/lock_test.go

```diff
@@ -96,143 +65,156 @@ func (suite *KeeperTestSuite) TestGetPeriodLocks() {
 	suite.Require().Len(locks, 1)
 }
 
-func (suite *KeeperTestSuite) TestUnlockPeriodLockByID() {
-	suite.SetupTest()
-	now := suite.Ctx.BlockTime()
-	lockKeeper := suite.App.LockupKeeper
-
-	// initial check
-	locks, err := lockKeeper.GetPeriodLocks(suite.Ctx)
-	suite.Require().NoError(err)
-	suite.Require().Len(locks, 0)
-
-	// lock coins
-	addr1 := sdk.AccAddress([]byte("addr1---------------"))
-	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
-	suite.LockTokens(addr1, coins, time.Second)
-
-	// unlock lock just now
-	lock, err := lockKeeper.GetLockByID(suite.Ctx, 1)
-	suite.Require().NoError(err)
-	err = lockKeeper.UnlockMaturedLock(suite.Ctx, lock.ID)
-	suite.Require().Error(err)
-
-	// move start time to 1 second in the future.
-	suite.Ctx = suite.Ctx.WithBlockTime(now.Add(time.Second))
-
-	// Try to finish unlocking a lock, before starting unlock.
-	lock, err = lockKeeper.GetLockByID(suite.Ctx, 1)
-	suite.Require().NoError(err)
-	err = lockKeeper.UnlockMaturedLock(suite.Ctx, lock.ID)
-	suite.Require().Error(err)
-
-	// begin unlock
-	lock, err = lockKeeper.GetLockByID(suite.Ctx, 1)
-	suite.Require().NoError(err)
-	err = lockKeeper.BeginUnlock(suite.Ctx, lock.ID, nil)
-	suite.Require().NoError(err)
-
-	// unlock 1s after begin unlock
-	lock, err = lockKeeper.GetLockByID(suite.Ctx, 1)
-	suite.Require().NoError(err)
-	err = lockKeeper.UnlockMaturedLock(suite.Ctx.WithBlockTime(now.Add(time.Second*2)), lock.ID)
-	suite.Require().NoError(err)
-
-	// check locks
-	locks, err = lockKeeper.GetPeriodLocks(suite.Ctx)
-	suite.Require().NoError(err)
-	suite.Require().Len(locks, 0)
-}
-
 func (suite *KeeperTestSuite) TestUnlock() {
-	// test for coin unlocking
 	suite.SetupTest()
-	now := suite.Ctx.BlockTime()
+	initialLockCoins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
+
+	testCases := []struct {
+		name                          string
+		unlockingCoins                sdk.Coins
+		expectedBeginUnlockPass       bool
+		passedTime                    time.Duration
+		expectedUnlockMaturedLockPass bool
+		balanceAfterUnlock            sdk.Coins
+	}{
+		{
+			name:                          "normal unlocking case",
+			unlockingCoins:                initialLockCoins,
+			expectedBeginUnlockPass:       true,
+			passedTime:                    time.Second,
+			expectedUnlockMaturedLockPass: true,
+			balanceAfterUnlock:            initialLockCoins,
+		},
+		{
+			name:                          "begin unlocking with nil as parameter",
+			unlockingCoins:                nil,
+			expectedBeginUnlockPass:       true,
+			passedTime:                    time.Second,
+			expectedUnlockMaturedLockPass: true,
+			balanceAfterUnlock:            initialLockCoins,
+		},
+		{
+			name:                          "unlocking coins exceed what's in lock",
+			unlockingCoins:                sdk.Coins{sdk.NewInt64Coin("stake", 20)},
+			expectedBeginUnlockPass:       false,
```

Comment:
nit: if boolean is not initialized, it is false by default so no need to specify
```suggestion
```